File: python/kode program/hewan.py
from makhluk_hidup import MakhlukHidup

# ============================================
# Kelas Hewan (turunan dari MakhlukHidup)
# ============================================
# Hewan.py

# Kelas turunan dari MakhlukHidup
class Hewan(MakhlukHidup):
    def __init__(self, id, nama, umur, makanan, habitat):
        super().__init__(id, nama, umur)
        self.makanan = makanan
        self.habitat = habitat

    def tampilkan(self):
        # Override untuk menampilkan atribut khusus Hewan
        return f"[Hewan] {super().tampilkan()} {self.makanan:<12} {self.habitat:<12}"

File: java/kode program/Hewan.java
// Subclass Hewan, turunan dari MakhlukHidup
public class Hewan extends MakhlukHidup {
    private String makanan;
    private String habitat;

    public Hewan(int id, String nama, int umur, String makanan, String habitat) {
        super(id, nama, umur);
        this.makanan = makanan;
        this.habitat = habitat;
    }

    @Override
    public String getInfo() {
        return String.format("Hewan   | ID: %d | Nama: %s | Umur: %d | Makanan: %s | Habitat: %s",
                this.id, this.nama, this.umur, this.makanan, this.habitat);
    }
}
